### DISCUSSION TITLE
UBL probe values incorrect

### DISCUSSION THREAD

**Discussion by bocca86** [Jyers/Marlin#643] [Q&A] [answered]
i cannot get UBL to work for some reason the mesh points are always off.
the bed is perfectly level using the manual paper method if i print without ubl mesh loaded i get decent prints.
but with the mesh enabled one corner always seems to be way too low first it was top left reflashed the firmware and now it's bottom left.
using a BL touch wich was working fine before switching from smith3D to jyers UBL

**Comment by RPhilbo** [comment]
Is your Probe Offset set to the correct values?

Maybe you should get a look here:
https://github.com/Jyers/Marlin/discussions/461#discussioncomment-664481

**Reply by bocca86** [reply]
They should be, i set the measured offsets in the menu and just checked using M851 it replied the correct values

**Comment by Jyers** [comment, marked as answer]
Are you still facing this issue? It could perhaps be a hardware issue of some kind. Nothing about the leveling processing would have changed between firmwares.

**Reply by bocca86** [reply]
i am still having the same problem, i'm not excluding hardware issues but i can't figure out what it would be.
if it's a slanted x-gantry it wouldnt just be the bottom of the bed but the entire left side if it's the bed it wouldnt be level and print fine without UBL and the bl-touch was working fine before

**Reply by gothcha** [reply]
What does the mesh look like in the viewer? Does it show a higher offset in the affected area? How far exactly in mm is the z height out? 
I found that the left 10-15mm side of my mesh just beyond the probed points was showing 0.2mm when the majority of the bed was practically zero. I suspected that the interpolation of the remaining points was not accurate (there is no way the code can extend the mesh without any data). I modified the probe margins and the mesh insets such that all 100 mesh points were probed. Then I found that the left side was now in line with expectations.

**Reply by gothcha** [reply]
These were the settings I used in Configuration_adv.h

#define MESH_MIN_X 0
#define MESH_MIN_Y 10
#define MESH_MAX_X X_BED_SIZE - 23
#define MESH_MAX_Y Y_BED_SIZE - 15

#define PROBING_MARGIN_LEFT 0
#define PROBING_MARGIN_RIGHT 23
#define PROBING_MARGIN_FRONT 10
#define PROBING_MARGIN_BACK 15

The back margin and mesh needs to greater than the Y Probe offset (-11 in my case) and the right needs to be set from the X probe offset (-42) and the max X travel (249). Travel max (249) - probe offset (42) = 207 so X_BED_SIZE - _23_ = 207

This gave all the left side probed accurately and sufficient margin the avoid the clips at the front and back.

**Comment by apr1967** [comment]
I switched to rubber springs and it seems better.
